sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -47,7 +49,10 @@ public Response Response
                 if (_response == null)
                 {
 #pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
-                    throw new InvalidOperationException("Response was not set, make sure SendAsync was called");
+                    throw new InvalidOperationException("Response is not set on this message. "
+                        + "This is may be because the message was not sent via pipeline.Send, "
+                        + "the pipeline transport did not populate the response, or because "
+                        + "TransferResponseDisposeOwnership was called.");
```

Comment:
you might want to use nameof() instead of hardcoding the method name.

Reply:
Good suggestion